pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -1081,6 +1192,52 @@ private static MutableRoaringBitmap getQueryableDocIdsSnapshotFromSegment(IndexS
     return queryableDocIdsSnapshot;
   }
 
+  private void setSegmentContexts(List<SegmentContext> segmentContexts) {
+    for (SegmentContext segmentContext : segmentContexts) {
+      IndexSegment segment = segmentContext.getIndexSegment();
+      if (_trackedSegments.contains(segment)) {
+        segmentContext.setQueryableDocIdsSnapshot(getQueryableDocIdsSnapshotFromSegment(segment));
+      }
+    }
+  }
+
+  private boolean skipUpsertViewRefresh(long upsertViewFreshnessMs) {
+    long nowMs = System.currentTimeMillis();
```

Comment:
(minor) Check `upsertViewFreshnessMs` before getting the current time. We may also check for `== 0` and directly return `false`